### ISSUE TITLE
to_json_string to ignore null field

### ISSUE THREAD

**Issue by snigdhasjg** [confluentinc/ksql#9214]
**Is your feature request related to a problem? Please describe.**
Sometimes `null` field doesn’t add any value. For example we sink kafka topic to postgres `JSONB` column, which will add up unnecessary storage.

Requesting to provide optional flag to ignore null value.

_Note: We can also consider similar for empty value as well_

**Describe the solution you'd like**
Function that we have today: `to_json_string(STRUCT<> obj)`

We can have overloaded method: `to_json_string(STRUCT<> obj, Boolean includeNulls)`
Where `includeNulls` default value is `true` to match current behaviour

It shouldn’t be server config as to support different use-cases.

**Describe alternatives you've considered**
N/A

**Additional context**
[ToJsonString.java][1] internally uses ObjectMapper via [UdfJsonMapper.java][2]. 
And KSQL uses 1 static object of ObjectMapper.

One way to do it via `objectMapper.setSerializationInclusion(Include.NON_NULL);`. But doing so it enable it global level.
Not sure how we can do it on `STRUCT<>` level.

[1]:https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/ToJsonString.java
[2]:https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/UdfJsonMapper.java

**Comment by Gerrrr** [comment]
**TL;DR**

* @snigdhasjg you can achieve desired behavior by using `map` as a container for your data instead of `struct` and filtering out `null` values using [filter](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/scalar-functions/#filter). See the snippet below. Alternatively, you can get similar behavior with `struct` and `json_records`, but you'll lose type information in the JSON representation:

```sql
ksql> select to_json_string(filter(json_records(to_json_string(struct(a := 1, b:= 2, c := ''))), (k, v) => v != '')) from s1;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|KSQL_COL_0                                                                                                                                                       |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|{"A":"1","B":"2"}                                                                                                                                                |
Query Completed
Query terminated
```

* we might want to allow creating `struct` with a new type from an existing `struct` and a function, similar to `transform/filter/reduce`. I am not sure if it is possible with our type system though. See the reasoning below.

---

Requested use case is very much valid, but I propose to approach it from a different angle. Do we need these filtering options for `to_json_string` only in the case of `struct` or is it generally useful?

Consider the same use case, but the `obj` is a `map` instead of a `struct`:

```sql
ksql> select to_json_string(map('a' := 1, 'b':= 2, 'c' := null)) from s1;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|KSQL_COL_0                                                                                                                                                       |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|{"a":1,"b":2,"c":null}                                                                                                                                           |
Query Completed
Query terminated
```

Then, we could have achieved desired behavior by filtering out `null` values before using `to_json_string`:

```SQL
ksql> select to_json_string(filter(map('a' := 1, 'b':= 2, 'c' := null), (k, v) => v IS NOT NULL)) from s1;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|KSQL_COL_0                                                                                                                                                       |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
|{"a":1,"b":2}                                                                                                                                                    |
Query Completed
Query terminated
```

FWIW we can do a similar transformation for arrays. The "filter out `null` values" use case is only unsupported for structs and adding an argument to `to_json_string` seems to be a workaround for very specific case we might prefer to avoid. Instead, we might want to make it possible to create a new struct from an existing struct and a function. Something like:

```transform(struct<...>, (s => ...) => struct<...>```

On the other hand, I am not sure if our type system supports a return type that will be determined in runtime.
